libs/common/src/platform/abstractions/crypto.service.ts

```
@@ -239,8 +253,22 @@ export abstract class CryptoService {
    * Returns the private key from memory. If not available, decrypts it
    * from storage and stores it in memory
    * @returns The user's private key
+   *
+   * @throws An error if there is no user currently active.
+   *
+   * @deprecated Use {@link userPrivateKey$} instead.
    */
   abstract getPrivateKey(): Promise<Uint8Array>;
+
+  /**
+   * Gets an observable stream of the given users decrypted private key, will emit null if the user
+   * doesn't have a UserKey to decrypt the encrypted private key or null if the user doesn't have an
+   * encrypted private key at all.
+   *
+   * @param userId The user id of the user to get the data for.
+   */
+  abstract userPrivateKey$(userId: UserId): Observable<UserPrivateKey>;
+
```

same question here about return type: `Observable<UserPrivateKey | null>`

I'm also wondering if we want to return an error if the user doesn't have an encrypted private key, but does have a user key. We've talked about setting both of those together in the future, which would give us the assurance that we're not simply somewhere between the UserKey being set and the PrivateKey being set.